test/unit/test_builder.py

```diff
@@ -1258,3 +1312,69 @@ def test_compose_status_retry(self):
 
         client = self.plugin.Client("http://localhost")
         client.wait_for_compose(compose_id, sleep_time=0.1)
+
+
+@PluginTest.load_plugin("builder")
+class TestUploadOptionsRegistry(PluginTest):
+
+    def test_no_options(self):
+        """Test that no options are returned if no upload options are set"""
+        reg = self.plugin.UploadOptionsRegistry.from_data(None)
+        for c in itertools.product(["x86_64", "s390x"], ["aws", "azure", "gce", "guest-image"]):
+            self.assertEqual(reg.get_options(c[0], c[1]), None)
+
+    def test_global_options(self):
+        """Test that global options are returned for any image-type / arch combination"""
+        global_options = {
+            "region": "us-east-1",
+            "share_with_accounts": ["123456789"]
+        }
+        reg = self.plugin.UploadOptionsRegistry.from_data(global_options)
+        for c in itertools.product(["x86_64", "s390x"], ["aws", "azure", "gce", "guest-image"]):
+            self.assertEqual(reg.get_options(c[0], c[1]), global_options)
+
+    def test_combination_specific_options(self):
+        """Test that upload options are returned only for defined combinations of image-type and arch"""
+
+        options = [
+            {
+                "arch": "x86_64",
+                "image_type": "aws",
+                "upload_options": {
+                    "region": "us-east-1",
+                    "share_with_accounts": ["123456789"]
+                }
+            },
+            {
+                "arch": "aarch64",
+                "image_type": "aws",
+                "upload_options": {
+                    "region": "us-east-123",
```

Comment:
East US is now completely filled with AWS datacenters. 🤭

Reply:
I know... those datacenters, right? 🤣